docs/getting-started/publish-infrastructure.md

```diff
@@ -0,0 +1,560 @@
+---
+title: Publish Infrastructure
+toc: true
+weight: 4
+indent: true
+---
+
+# Publish Infrastructure
+
+Provisioning infrastructure using the Kubernetes API is a powerful capability,
+but combining primitive infrastructure resources into a single unit and
+publishing them to be provisioned by developers and consumed by applications
+greatly enhances this functionality.
+
+As mentioned in the [last section], CRDs that represent infrastructure resources
+on a provider are installed at the *cluster scope*. However, applications are
+typically provisioned at the *namespace scope* using Kubernetes primitives such
+as `Pod` or `Deployment`. To make infrastructure resources available to be
+provisioned at the namespace scope, they can be *published*. This consists of
+creating the following resources:
+
+- `InfrastructureDefinition`: defines a new kind of composite resource
+- `InfrastructurePublication`: makes an `InfrastructureDefinition` available at
+  the namespace scope
+- `ClusterRole`: allows Crossplane to manage this newly defined infrastructure
+  type
+- `Composition`: defines one or more resources and their configuration
+
+In addition to making provisioning available at the namespace scope,
+[composition] also allows for multiple types of managed resources to satisfy the
+same namespace-scoped resource. In the examples below, we will define and
+publish a new `MySQLInstance` resource that only takes a single `storageGB`
+parameter, and specifies that it will create a connection `Secret` with keys for
+`username`, `password`, and `hostname`. We will then create a `Composition` for
+each provider that can satisfy and be parameterized by a `MySQLInstance. Let's
+get started!
+
+## Create ClusterRole
+
+Before we defined new types we need to make sure that Crossplane is able to
+manage them for us. Creating a `ClusterRole` with the
+`rbac.crossplane.io/aggregate-to-crossplane: "true"` and the `MySQLInstance` and
+`MySQLInstanceRequirement` with `*` verbs will make that possible.
+
+Create a file named `mysql-cr.yaml` with the following contents:
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: mysqlinstances.database.example.org
+  labels:
+    rbac.crossplane.io/aggregate-to-crossplane: "true"
+rules:
+- apiGroups:
+  - database.example.org
+  resources:
+  - mysqlinstances
+  - mysqlinstances/status
+  - mysqlinstancerequirements
+  - mysqlinstancerequirements/status
+  verbs:
+  - "*"
+```
+
+Create the `ClusterRole`:
+```
+kubectl apply -f mysql-cr.yaml
+```
```

Comment:
Could we store this `ClusterRole` as `quick-start-role.yaml` or something and `kubectl apply -f https://raw.url/quick-start-role.yaml`, without showing its contents here. Obviously this step needs to happen for things to work, but I don't think we add much value by demonstrating RBAC in motion.

Reply:
I personally feel like it’s kind of weird to see everything then just `kubectl apply` this one thing. That being said, it is not a core concept and I get that the idea is simplicity here. It seems like we are already assuming folks are familiar with k8s concepts though so maybe it’s not adding much complexity?

Reply:
If I understand correctly the purpose of this guide is take folks on a tour of Crossplane concepts, getting them up and running as quickly as possible. This `ClusterRole` is needed for the Crossplane concepts we're introducing to work, but it's not a Crossplane concept, so to me it doesn't seem worth the extra lines of YAML and cognitive burden on the reader. We show the ClusterRole inline and discuss it in more detail in the longer composition guide.

Given this is a quick start I'd err toward including an obvious but succinct note like:

> Note that Crossplane must be granted an RBAC role allowing it to manage your newly defined and published infrastructure. Run `kubectl apply -f https://github.com/quick-start-role.yaml` to grant Crossplane access to the resources we'll cover in this guide.

Reply:
That sounds reasonable to me! Will update 👍